app/domain/authentication/authn_ldap/authenticator.rb

```diff
@@ -7,10 +7,14 @@
 module Authentication
   module AuthnLdap
 
-    class Authenticator
+    class Authenticator < ::Dry::Struct
+      attr_reader :role_cls
+      attr_reader :credentials_cls
```

Comment:
What's using these `attr_reader`s? Or is this just something you have to specify for `::Dry::Struct`? If not, I would omit them and refer to the variables using `@` notation within the class.

Reply:
@jtuttle You need them to make the object constructor.  As I explained to Micah (and may have forgotten to explain to you), my use of `Dry::Struct` for these command objects was a clear error and abuse of `Dry::Struct` in retrospect.  `Dry::Struct` should be reserved for value objects only.  The correct solution here is to rewrite these old objects with `CommandClass` or even vanilla ruby.

Reply:
As mentioned in slack, the inheritance from `::Dry::Struct` should be removed.